subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dsl/dependencies/DefaultDependencyHandler.java

```diff
@@ -229,6 +230,7 @@ public void artifactTypes(Action<? super ArtifactTypeContainer> configureAction)
 
     @Override
     public void registerTransform(Action<? super VariantTransform> registrationAction) {
+        DeprecationLogger.nagUserOfDeprecated("Using the ArtifactTransform API", "Use TransformAction instead.");
```

Comment:
I think this message might be a bit misleading, because reading the first sentence sounds like artifact transforms in general have been deprecated.

Why don't we say something like:

```suggestion
        DeprecationLogger.nagUserOfDeprecated("Registering artifact transforms implementing ArtifactTransform", "Implement TransformAction instead.");
```